### PR TITLE
CONSOLE-3005: Provide alias for dynamic plugin's proxy

### PR DESCRIPTION
**Pull request by jhadvig**
Story: https://issues.redhat.com/browse/CONSOLE-3005

Changes are based on the [API change PR](https://github.com/openshift/api/pull/1067)

/assign @christianvogt @spadgett

### CODE REVIEW

File: go.mod
Comment (jhadvig) [inline go.mod:29]: once https://github.com/openshift/api/pull/1067 will get merged will update both pkgs... api && client-go

**Comment by jhadvig** [comment]
QE Approver:
 /assign @yapei
Docs Approver: 
/assign @opayne1
PX Approver: 
/assign @RickJWagner
Approver:
 /assign @spadgett

**Review by spadgett — APPROVED** [review]
/approve

Changes LGTM, just need to wait for the API PR to merge

**Comment by yapei** [comment]
1. deploy https://raw.githubusercontent.com/yapei/console/780ef7276e1bda53fd94c419d985e43931badf90/dynamic-demo-plugin/oc-manifest.yaml(replace image with quay.io/yapei/console-demo-plugin-separator-story:latest) which includes ConsolePlugin with following spec
```
apiVersion: console.openshift.io/v1alpha1
kind: ConsolePlugin
metadata:
  name: console-demo-plugin
spec:
  displayName: 'OpenShift Console Demo Plugin'
  service:
    name: console-demo-plugin
    namespace: console-demo-plugin
    port: 9001
    basePath: '/'
  proxy:
  - type: Service
    alias: demoplugin
    service:
        name: thanos-querier
        namespace: openshift-monitoring
        port: 9091
        authorize: true
```
$ oc apply -f oc-manifest.yaml
2. enable `console-demo-plugin` and visit `/test-proxy-service` URL, response of request `/api/proxy/console-demo-plugin/demoplugin/api/v1/rules` is displayed on the page
Screen Shot 2022-01-11 at 4 48 21 PM

**Pull request merged**
